gnomad_qc/v4/annotations/generate_freq_genomes.py

```diff
@@ -1366,35 +1425,70 @@ def main(args):
+            "Annotating call stats group membership for pop diff release samples..."
+        )
+        # Need to make sure we are using the names that are in the vds for the release
+        # samples.
+        # Note: the samples in the pop_diff HT are samples in the to-be-split 'Han' and
+        # 'Papuan' populations AND their 'gnomad_release' status hasn't changed.
+        v3_sites_meta_rename_ht = v3_sites_meta_ht.key_by(
+            s_no_prefix=v3_sites_meta_ht.s.replace("v3.1::", "")
+        )
+        pop_diff_sample_ht = get_updated_release_samples(res.updated_meta_ht.ht())[0]
+        pop_diff_sample_ht = pop_diff_sample_ht.key_by(
+            s=v3_sites_meta_rename_ht[pop_diff_sample_ht.s].s
+        )
+        pop_diff_group_membership_ht = get_group_membership_ht_for_an(
+            pop_diff_sample_ht, only_pop_diff=True
+        )
+        pop_diff_group_membership_ht = pop_diff_group_membership_ht.checkpoint(
+            new_temp_file("group_membership_pop_diff", "ht")
+        )
+        logger.info(
+            "Computing the AN HT of v3.1 pop diff samples for new v4.0 genomes"
+            " variants. Freq meta: %s",
+            hl.eval(pop_diff_group_membership_ht.freq_meta),
+        )
+        ht = compute_an_by_group_membership(
+            v3_vds, pop_diff_group_membership_ht, res.freq_join_ht.ht().select()
         )
-        ht = compute_an_by_group_membership(v3_vds, group_membership_ht, ht)
-        ht.write(res.v3_release_an_ht.path, overwrite=overwrite)
+        ht.write(res.v3_pop_diff_an_ht.path, overwrite=overwrite)
+
+        if not args.pop_diff_an_only:
+            logger.info(
+                "Annotating call stats group membership for each v3.1 release sample..."
+            )
+            v3_sites_meta_ht = v3_vds.variant_data.cols()
+            v3_sites_meta_ht = v3_sites_meta_ht.filter(v3_sites_meta_ht.meta.release)
+            group_membership_ht = get_group_membership_ht_for_an(v3_sites_meta_ht)
+            group_membership_ht = group_membership_ht.checkpoint(
+                new_temp_file("group_membership_all", "ht")
+            )
+
+            logger.info(
+                "Computing the AN HT of v3.1 samples for new v4.0 genomes variants..."
+            )
+            ht = compute_an_by_group_membership(v3_vds, group_membership_ht, ht)
+            ht.write(res.v3_release_an_ht.path, overwrite=overwrite)
 
     if args.update_release_callstats:
         res = v4_genome_release_resources.update_release_callstats
         res.check_resource_existence()
 
         logger.info("Merging all call stats HTs for final v4.0 genomes call stats...")
         ht = generate_v4_genomes_callstats(
-            res.freq_join_ht.ht(), res.v3_release_an_ht.ht()
+            res.freq_join_ht.ht(), res.v3_release_an_ht.ht(), res.v3_pop_diff_an_ht.ht()
         )
         ht = get_histograms(ht, v3_sites_ht)
         ht = ht.annotate_globals(
+            downsamplings=v3_sites_ht.downsamplings,
```

Comment:
There's no `downsamplings` in the globals in v3 release HT.

Reply:
How about we just make a list of `downsampling` from the `freq_meta`?

Reply:
yeah, I like making a list from freq_meta